libsupport/src/EntityTypeManager.cpp

```diff
@@ -1,5 +1,40 @@
 #include "katana/EntityTypeManager.h"
 
+#include "katana/Logging.h"
+#include "katana/Result.h"
+
+size_t
+katana::EntityTypeManager::CalculateSetOfEntityTypeIDsSize(
+    EntityTypeID max_id) {
+  // min number of bits to fit bitset[max_id] is max_id +1
+  size_t min_size = max_id + 1;
+
+  KATANA_LOG_VASSERT(
+      max_id < kInvalidEntityType, "Katana only supports {} entity types",
+      kInvalidEntityType);
+
+  size_t new_size = kDefaultSetOfEntityTypeIDsSize;
+  // Double the default size of the set until it is bigger than min_size.
+  // There are faster, more clever ways to do this, like using a DeBruijn sequence but
+  // 1) we don't use this function too often
+  // 2) this is easier to understand
+  while (new_size < min_size) {
```

Comment:
This logic is redundant. `PODVector` in `DynamicBitsetSlow` handles this (by using capacity and size similar to std::vector)

Reply:
While it will reserve the extra space ahead of time, making the call to resize faster, it won't actually resize larger than you ask. 
PodVector.h
```
  void resize(size_t n) {
    reserve(n);
    size_ = n;
  }
```
The slow part that I'm trying to avoid here is calling resize in the first place. 
When we add an EntityTypeID, we have to go through at most 2x (number of EntityTypeID) bitsets and resize each one.
Having to call resize on 130k bitsets would be much slower than resizing early, and avoiding the calls.

Reply:
That said, maybe the optimization isn't worth the extra code/work?

Reply:
Fair point. But setting the size should be reasonably cheap. Yes, I don't think the extra code is worth. However, I'd prefer we merge this soon, so it's fine to keep the code around. If and only if you push further changes, add a TODO/note in the code regarding this.